File: problems/3002/Solution.cs
public class Solution {
    public int MaximumSetSize(int[] nums1, int[] nums2) {
        var s1 = new System.Collections.Generic.HashSet<int>(nums1);
        var s2 = new System.Collections.Generic.HashSet<int>(nums2);
        var all = new System.Collections.Generic.HashSet<int>(s1);
        all.UnionWith(s2);
        int half = nums1.Length / 2;
        return System.Math.Min(all.Count, System.Math.Min(s1.Count, half) + System.Math.Min(s2.Count, half));
    }
}

File: problems/3002/solution.go
func maximumSetSize(nums1 []int, nums2 []int) int {
    s1, s2, all := map[int]struct{}{}, map[int]struct{}{}, map[int]struct{}{}
    for _, x := range nums1 { s1[x] = struct{}{}; all[x] = struct{}{} }
    for _, x := range nums2 { s2[x] = struct{}{}; all[x] = struct{}{} }
    half := len(nums1) / 2
    min := func(a, b int) int { if a < b { return a }; return b }
    return min(len(all), min(len(s1), half)+min(len(s2), half))
}
